File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void) 
{
    // Get height of pyramid from user.
    int height;
    const int MIN_HEIGHT = 1;
    const int MAX_HEIGHT = 8;
    do
    {
        height = get_int("Height: ");
    } while (height < MIN_HEIGHT || height > MAX_HEIGHT);

    // Print a right-aligned pyramid and a left-aligned pyramid.
    int spaces = height - 1;
    for (int row = 0; row < height; row++)
    {
        for (int col = 0; col < height; col++)
        {
            if (col < spaces)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }

        spaces--;
        printf("  ");
        
        for (int col = 0; col <= row; col++)
        {
            printf("#");
        }
        
        printf("\n");
    }
    
    return 0;
}

File: sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = get_height()
    print_pyramid(height)


def get_height():
    height = get_int('Height: ')

    while height < 1 or height > 8:
        height = get_int('Height: ')
    
    return height


def print_pyramid(height):
    spaces = height - 1

    for i in range(1, height + 1):
        print(' ' * spaces, end='')
        print('#' * i, end='')
        spaces -= 1
        print('  ', end='')
        print('#' * i)


if __name__ == '__main__':
    main()